### PR TITLE
User datasets community sharing updates

### LINKED ISSUE #1141: User datasets - Update styling and user flow in the sharing modal
(related issues that could be closed possibly: https://github.com/VEuPathDB/web-monorepo/issues/411 https://github.com/VEuPathDB/web-monorepo/issues/926)

The following mock-ups were generated from a UX discussion on 2024-07-15. They detail changes to the UI for sharing a user dataset.

**Manage study access dropdown**
image

**Community access**
image

**Community access - Access granted**
image

**Community access - Access revoked**
image

**Individual access**
image

**Individual access - shared**
image

### PR DESCRIPTION
**Pull request by dmfalke**
Fixes #1141

Changed include:

- Add community modal state to Redux store
- Dropdown button to choose between community and individual
- Use coreui modal
- Tweak verbiage

### CODE REVIEW

File: packages/libs/user-datasets/src/lib/Components/Sharing/UserDatasetCommunityModal.jsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+
+import { IconAlt as Icon, Loading } from '@veupathdb/wdk-client/lib/Components';
+import { WdkDependenciesContext } from '@veupathdb/wdk-client/lib/Hooks/WdkDependenciesEffect';
+import { Modal } from '@veupathdb/coreui';
+import { FilledButton } from '@veupathdb/coreui';
+
+import './UserDatasetSharingModal.scss';
+
+class UserDatasetSharingModal extends React.Component {
```
Comment (jernestmyers) [inline packages/libs/user-datasets/src/lib/Components/Sharing/UserDatasetCommunityModal.jsx:10]: Is there a particular reason you went with a class component here?

File: packages/libs/user-datasets/src/lib/Components/ThemedDeleteButton.tsx [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import { useUITheme } from '@veupathdb/coreui/lib/components/theming';
 import { MesaButton, Trash } from '@veupathdb/coreui';
 import { gray, mutedRed } from '@veupathdb/coreui/lib/definitions/colors';
-import { ThemedButtonProps } from './ThemedGrantAccessButton';
 
-export function ThemedDeleteButton({ buttonText, onPress }: ThemedButtonProps) {
+export interface Props {
```
Comment (jernestmyers) [inline packages/libs/user-datasets/src/lib/Components/ThemedDeleteButton.tsx:5]: Nit: Does this need to be exported?

File: packages/libs/user-datasets/src/lib/Components/ThemedGrantAccessButton.tsx [hunk 3]
```diff
@@ -1,53 +1,99 @@
 import { useUITheme } from '@veupathdb/coreui/lib/components/theming';
-import { MesaButton, Share } from '@veupathdb/coreui';
+import { colors, MesaButton, Share, SingleSelect } from '@veupathdb/coreui';
+import { PartialButtonStyleSpec } from '@veupathdb/coreui/lib/components/buttons';
+import { useMemo } from 'react';
 
-export type ThemedButtonProps = {
+export interface Props {
```
Comment (jernestmyers) [inline packages/libs/user-datasets/src/lib/Components/ThemedGrantAccessButton.tsx:6]: Nit: same as above re: export

**Review by jernestmyers — APPROVED** [review]
LGTM!

Note: I wasn't able to play with it on a dev site due to QA sites being down currently.

**Comment by dmfalke** [comment]
@jernestmyers thanks for the review! You're comments are great, and I will address them shortly. I did a bit of copy-paste here :disappointed:

**Pull request merged**